cli/config/configfile/file.go

```diff
@@ -36,6 +36,10 @@ type ConfigFile struct {
 	Filename             string                      `json:"-"` // Note: for internal use only
 	ServiceInspectFormat string                      `json:"serviceInspectFormat,omitempty"`
 	ServicesFormat       string                      `json:"servicesFormat,omitempty"`
+	HTTPProxy            string                      `json:"httpProxy,omitempty"`
+	HTTPSProxy           string                      `json:"httpsProxy,omitempty"`
+	NoProxy              string                      `json:"noProxy,omitempty"`
+	FTPProxy             string                      `json:"ftpProxy,omitempty"`
```

Comment:
yea, I tend to view this less of a go or json convention and more one of consistency with how they would specify things elsewhere.  For example, `--build-arg http_PROXY=...` should probably not be the same a `--build-arg HTTP_PROXY=...` since those really are two different env vars. So we should treat them as different in the config file too (assuming go allows us to - it might try to be "helpful" here and treat them the same :-( ).  But we should document/code it the right way.